variables.tf

```diff
@@ -1,154 +1,474 @@
-variable "subnet_type_tag_key" {
+variable "vpc_id" {
   type        = string
-  default     = "cpco.io/subnet/type"
-  description = "Key for subnet type tag to provide information about the type of subnets, e.g. `cpco.io/subnet/type=private` or `cpco.io/subnet/type=public`"
+  description = "VPC ID where subnets will be created (e.g. `vpc-aceb2723`)"
 }
 
-variable "subnet_type_tag_value_format" {
-  default     = "%s"
-  description = "This is using the format interpolation symbols to allow the value of the subnet_type_tag_key to be modified."
-  type        = string
+variable "igw_id" {
+  type        = list(string)
+  description = <<-EOT
+    The Internet Gateway ID that the public subnets will route traffic to.
+    Used if `public_route_table_enabled` is `true`, ignored otherwise.
+    EOT
+  default     = []
+  validation {
+    condition     = length(var.igw_id) < 2
+    error_message = "Only 1 igw_id can be provided."
+  }
+}
+
+variable "ipv6_egress_only_igw_id" {
+  type        = list(string)
+  description = <<-EOT
+    The Egress Only Internet Gateway ID the private IPv6 subnets will route traffic to.
+    Used if `private_route_table_enabled` is `true` and `ipv6_enabled` is `true`, ignored otherwise.
+    EOT
+  default     = []
+  validation {
+    condition     = length(var.ipv6_egress_only_igw_id) < 2
+    error_message = "Only 1 ipv6_egress_only_igw_id can be provided."
+  }
 }
 
 variable "max_subnet_count" {
+  type        = number
+  description = <<-EOT
+    Sets the maximum number of each type (public or private) of subnet to deploy.
+    0 will reserve a CIDR for every Availability Zone (excluding Local Zones) in the region, and
+    deploy a subnet in each availability zone specified in `availability_zones` or `availability_zone_ids`,
+    or every zone if none are specified. We recommend setting this equal to the maximum number of AZs you anticipate using,
+    to avoid causing subnets to be destroyed and recreated with smaller IPv4 CIDRs when AWS adds an availability zone.
+    Due to Terraform limitations, you can not set `max_subnet_count` from a computed value, you have to set it
+    from an explicit constant. For most cases, `3` is a good choice.
+    EOT
   default     = 0
-  description = "Sets the maximum amount of subnets to deploy. 0 will deploy a subnet for every provided availablility zone (in `availability_zones` variable) within the region"
 }
 
-variable "vpc_id" {
-  type        = string
-  description = "VPC ID where subnets will be created (e.g. `vpc-aceb2723`)"
+variable "max_nats" {
+  type        = number
+  description = "Maximum number of NAT Gateways or NAT instances to create"
+  default     = 999
```

Comment:
why ` 999`? should it be equal to the number of subnets or AZs?

Reply:
Really the default should be `MAX_INT` but Terraform doesn't provide such a value, so I used 999 as a "big enough" number that it will never take effect. It needs to be a value known at plan time, so it cannot be dependent on how many AZs are available.